### PR TITLE
Tagless auth

### PR DESCRIPTION
**Pull request by cquiroz**
This PR revamps the auth code with the following improvements:
* Update to the latest jwt library
* Replace argonaut for circe
* Use tagless style

The code could have used a bit more improvements (Some bits were use to learn about free monads) but given time constraints I'll leave it as is

### CODE REVIEW

File: modules/seqexec/web/server/src/main/scala/seqexec/web/server/http4s/WebServerLauncher.scala [hunk 1]
```diff
@@ -137,7 +137,7 @@ object WebServerLauncher extends IOApp with LogInitialization {
              /_/
 """
     val msg = s"""Start web server for site ${conf.site} on ${conf.mode} mode, version ${OcsBuildInfo.version}"""
-    L.info(banner + msg)
+    Logger[F].info(banner + msg)
```
Comment (jluhrs) [inline modules/seqexec/web/server/src/main/scala/seqexec/web/server/http4s/WebServerLauncher.scala]: Why this change? Doesn't `L` still exist in this context?

Comment (cquiroz) [reply modules/seqexec/web/server/src/main/scala/seqexec/web/server/http4s/WebServerLauncher.scala]: I thought about that, L is `Logger[IO]` and I wanted `Logger[F]`
